tx-pool/src/chunk_process.rs

```diff
@@ -0,0 +1,234 @@
+                        TimeRelativeTransactionVerifier::new(&rtx, snapshot.as_ref(), &tx_env)
+                            .verify()
+                            .map(|_| *completed)
+                            .map_err(Reject::Verification)?;
+
+                    let entry = TxEntry::new(rtx, completed.cycles, fee, tx_size);
+                    self.handle.block_on(
+                        self.service
+                            .submit_entry(completed, tip_hash, entry, status),
+                    )?;
+                    self.handle
+                        .block_on(self.service.after_process(tx, remote, &Ok(completed)));
+                    self.handle.block_on(self.remove_front());
+                    return Ok(false);
+                }
+                CacheEntry::Suspended(suspended) => {
+                    init_snap = Some(Arc::clone(&suspended.snap));
+                }
+            }
+        }
+
+        let data_loader = snapshot.as_data_provider();
+        let fee =
+            ContextualWithoutScriptTransactionVerifier::new(&rtx, consensus, &data_loader, &tx_env)
+                .verify()
+                .map_err(Reject::Verification)?;
+        let script_verifier = ScriptVerifier::new(&rtx, consensus, &data_loader, &tx_env);
+        let mut tmp_state: Option<ScriptVerifyState> = None;
+
+        let max_cycles = if let Some((declared_cycle, _peer)) = remote {
+            declared_cycle
+        } else {
+            consensus.max_block_cycles()
+        };
+        let current_cycles = init_snap.as_ref().map(|s| s.current_cycles).unwrap_or(0);
+
+        let completed: Completed = loop {
+            // Should get here until there is no command, otherwise there maybe have a very large delay
+            while let Ok(cmd) = self.recv.try_recv() {
+                match cmd {
+                    Command::Suspend => {
+                        let state = tmp_state.take();
+                        if let Some(state) = state {
+                            let snap = state.try_into().map_err(Reject::Verification)?;
+                            let txs_verify_cache = Arc::clone(&self.service.txs_verify_cache);
+                            self.handle.block_on(async move {
+                                let mut guard = txs_verify_cache.write().await;
+                                guard.put(tx_hash, CacheEntry::suspended(Arc::new(snap), fee));
+                            })
+                        }
+                        self.p_state = ProcessState::Interrupt;
+                        return Ok(false);
+                    }
+                    Command::Stop => {
+                        return Ok(true);
+                    }
+                    Command::Continue => {
+                        self.p_state = ProcessState::Normal;
+                    }
+                }
+            }
+
+            let step_cycle = cmp::min(MIN_STEP_CYCLE, max_cycles - current_cycles);
```

Comment:
It is necessary to judge whether it is 0. If it is 0, it should jump out. Although the probability is very small, but if it is 0, it will not be able to exit without the `Suspend` command.